### DISCUSSION TITLE
Control auto resample during zoom operation

### DISCUSSION THREAD

**Discussion by smartstoic** [highfestiva/finplot#524] [General]
Hello,

Thanks for the great work!

I plot 1 min OCHL data from a dataframe with index being the time in the following format: "2024-06-14 22:55:00"..

When I zoom in and out, I see that resampling is happening automatically. This is fantastic but some times I need to control it.

Is there a way to activate, deactivate this behavior? In some of your examples, this doesn't happen but I couldn't figure out how it works.

Thanks!
SS

**Comment by highfestiva** [comment]
The control isn't great atm, but you can do this:
```python3
plot = candlestick_ochl(...)
plot.resamp = None
fplt.show()
```

**Comment by smartstoic** [comment]
Perfect thanks!

A follow up question on the automatic resampling it self: is there any way to control the timeframes?

Also, is there a way to access the currently used timeframe and print it for information purposes on the plot?

**Reply by smartstoic** [reply]
I see that your resampling is based on a rolling sampling_rate. This makes the candles somewhat random in terms of timeframe. What I had in mind was predefine a few relevant timeframes with period higher that the df to be plotted eg for 15m dataframe rasample to 1M, 1W, 1D, 4H, 1H ... And then when replotting, select appropriate frame instead of creating random candles. This would be towards accuracy of the figure. For backtesting and visual analysis, this would be very useful.

**Reply by highfestiva** [reply]
Good idea, probably a visualization of the currently used resample period would be useful too in that scenario. My focus has mostly been on making the candle plot performant and ok to easily overview when zooming out on large datasets. If you're actually developing a backtesting strategy for distinct time frames, you probably should use a different approach. But if you come up with a good way to implement it, I'm all for it and will integrate your pull request.
